src/coreclr/src/vm/interoplibinterface.cpp

```diff
@@ -1198,6 +1247,78 @@ void ComWrappersNative::MarkExternalComObjectContextCollected(_In_ void* context
     }
 }
 
+void ComWrappersNative::MarkWrapperAsComActivated(_In_ IUnknown* wrapper)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        MODE_ANY;
+        PRECONDITION(wrapper != NULL);
+    }
+    CONTRACTL_END;
+
+    InteropLib::Com::MarkComActivated(wrapper);
+}
+
+void QCALLTYPE GlobalComWrappers::SetGlobalInstanceRegistered()
+{
+    QCALL_CONTRACT;
+
+    _ASSERTE(!g_IsGlobalComWrappersRegistered);
+
+    BEGIN_QCALL;
+
+    g_IsGlobalComWrappersRegistered = true;;
+
+    END_QCALL;
+}
+
+bool GlobalComWrappers::TryGetOrCreateComInterfaceForObject(
+    _In_ OBJECTREF instance,
+    _In_ INT32 flags,
+    _Outptr_ void** wrapperRaw)
+{
+    if (!g_IsGlobalComWrappersRegistered)
+        return false;
+
+    // Switch to Cooperative mode since object references
+    // are being manipulated.
+    {
+        GCX_COOP();
+
+        // Passing NULL as the ComWrappers implementation indicates using the globally registered instance
+        return TryGetOrCreateComInterfaceForObjectInternal(
+            NULL,
+            instance,
+            (CreateComInterfaceFlags)flags,
+            wrapperRaw);
+    }
+}
+
+bool GlobalComWrappers::TryGetOrCreateObjectForComInstance(
+    _In_ IUnknown* externalComObject,
+    _In_ INT32 flags,
+    _Out_ OBJECTREF* objRef)
+{
+    if (!g_IsGlobalComWrappersRegistered)
+        return false;
+
+    // Switch to Cooperative mode since object references
+    // are being manipulated.
+    {
+        GCX_COOP();
+
+        // Passing NULL as the ComWrappers implementation indicates using the globally registered instance
+        return TryGetOrCreateObjectForComInstanceInternal(
+            NULL /*comWrappersImpl*/,
+            externalComObject,
+            (CreateObjectFlags)flags,
+            true /*unwrap*/,
```

Comment:
Document the unwrapping logic here. Why is it being done in this call. Use cases should be in the documentation here or in the header.